### ISSUE TITLE
K8S CLI stays in "Connecting" in the Rancher UI

### ISSUE THREAD

**Issue by chrisyoung12345** [rancher/rancher#8467]
When attempting to access the Kubernetes CLI from the Rancher UI the screen simply states:

Or use this handy shell to directly execute kubectl commands:
Shell: kubernetes-kubectld-1
Connecting...

It never connects. However, if you click the "Generate Config" button and run locally everything works correctly.

---
Rancher	v1.5.3
Cattle	v0.177.10
User Interface	v1.5.8
Rancher CLI	v0.5.1
Rancher Compose	v0.12.4
Kubernetes v1.5.2-rancher1-3
Amazon Linux (Rancher Server Hosts)
Aurora DB
Rancher deployed in HA mode

**Comment by joshwget** [comment]
Can you show the logs of `kubectld`?

**Comment by chrisyoung12345** [comment]
What is the best command to execute to get you the information you are requesting? I can see the logs in the UI but those don't look overly useful. Attempting to access the information via the Rancher CLI is proving a bit more challenging. Just let me know what you want executed and I should be able to provide the information.

**Comment by chrisyoung12345** [comment]
Note: Per my prior note, `./rancher --host rancher-usw2a-host1 logs default/kubernetes`, for example, does not return any output.

**Comment by joshwget** [comment]
Can you run `rancher logs kubernetes-kubectld-1`?

**Comment by chrisyoung12345** [comment]
@joshwget We resolved this issue last night. Thank you and the team for the support. In this particular case, the issue was due to using HTTP and HTTPS protocols on the ELB when they needed to be set to TCP and SSL (Secure TCP) for the inbound connections for websockets to work properly. Once those changes were made, this issue went away and the CLI started working as did the CPU, Memory and other performance stats dashboards.

Thank you for the support and the fantastic team behind Rancher!
